scripts/build_powerplants.py

```diff
@@ -254,8 +254,29 @@ def add_custom_powerplants(ppl, inputs, config):
 
 
 def replace_natural_gas_technology(df):
-    mapping = {"Steam Turbine": "CCGT", "Combustion Engine": "OCGT"}
-    tech = df.Technology.replace(mapping).fillna("CCGT")
+    """
+    Maps and replaces gas technologies in the powerplants.csv onto model
+    compliant carriers.
+    """
+    mapping = {
+        "Steam Turbine": "CCGT",
+        "Combustion Engine": "OCGT",
+        "NG": "CCGT",
+        "Ng": "CCGT",
+        "NG/FO": "OCGT",
+        "Ng/Fo": "OCGT",
+        "NG/D": "OCGT",
+        "LNG": "OCGT",
+        "CCGT/D": "CCGT",
+        "CCGT/FO": "CCGT",
+        "LCCGT": "CCGT",
+        "CCGT/Fo": "CCGT",
+    }
+    df["Technology"] = df["Technology"].map(mapping).fillna("CCGT")
+    for tech in df["Technology"]:
+        if tech not in ["CCGT", "OCGT"]:
+            logger.info(tech, len(df[df["Technology"] == tech]))
+            df["Technology"] = df["Technology"].map({tech: "CCGT"})
```

Comment:
This may be improved in something like:
```python
unknown_techs = np.setdiff1d(df["Technology"].unique(), ["CCGT", "OCGT"])
if len(unknown_techs) > 0:
             logger.info("The following unrecognized natural gas power plants converted into CCGT: ", unknown_techs)
            df["Technology"] = df["Technology"].map({t:"CCGT" for t in unknown_techs})
```

That code may be improved. the fact is that the proposed for loop loops over each row and may be not so efficient.
On second thought, the value counts may be added into the logging, but it may be an overkill maybe for the average user.
That code above may need some revision though.

Reply:
> This may be improved in something like:
> 
> ```python
> unknown_techs = np.setdiff1d(df["Technology"].unique(), ["CCGT", "OCGT"])
> if len(unknown_techs) > 0:
>              logger.info("The following unrecognized natural gas power plants converted into CCGT: ", unknown_techs)
>             df["Technology"] = df["Technology"].map({t:"CCGT" for t in unknown_techs})
> ```
> 
> That code may be improved. the fact is that the proposed for loop loops over each row and may be not so efficient. On second thought, the value counts may be added into the logging, but it may be an overkill maybe for the average user. That code above may need some revision though.

Very neat! Let me test it out.